code/mission/missionparse.cpp

```diff
@@ -2457,12 +2457,23 @@ int parse_create_object_sub(p_object *p_objp)
 
 	// If the ship is in a wing, this will be done in mission_set_wing_arrival_location() instead
 	// If the ship is in a wing, but the wing is docked then addition of bool brought_in_docked_wing accounts for that status --wookieejedi
-	if (Game_mode & GM_IN_MISSION && ((shipp->wingnum == -1) || (brought_in_docked_wing))) {
-		object *anchor_objp = (anchor_objnum >= 0) ? &Objects[anchor_objnum] : nullptr;
+	if (Game_mode & GM_IN_MISSION) {
+
+		if (Ship_info[shipp->ship_info_index].is_big_or_huge()) {
+			float mission_time = f2fl(Missiontime);
+			int minutes = (int)(mission_time / 60);
+			int seconds = (int)mission_time % 60;
+
+			mprintf(("%s arrived at %02d:%02d\n", shipp->ship_name, minutes, seconds));
+		}
 
-		Script_system.SetHookObjects(2, "Ship", &Objects[objnum], "Parent", anchor_objp);
-		Script_system.RunCondition(CHA_ONSHIPARRIVE, &Objects[objnum]);
-		Script_system.RemHookVars({"Ship", "Parent"});
+		if ((shipp->wingnum == -1) || (brought_in_docked_wing)) {
+			object* anchor_objp = (anchor_objnum >= 0) ? &Objects[anchor_objnum] : nullptr;
+
+			Script_system.SetHookObjects(2, "Ship", &Objects[objnum], "Parent", anchor_objp);
+			Script_system.RunCondition(CHA_ONSHIPARRIVE, &Objects[objnum]);
+			Script_system.RemHookVars({"Ship", "Parent"});
+		}
```

Comment:
This isn't the right way to do it.  First, that is one conceptually complete condition; separating it up like you did doesn't really make sense.  Second, there are multiple places where a ship can arrive and this is just one place.

Put the condition back to the way it was.  If you want to print your log message along with the script hook, this makes sense, but you will need to do it all four places the script hook appears.  Alternatively, you could put your mprintf statements inside `mission_log_add_entry`.

Reply:
I considered putting it in `mission_log_add_entry` but it is very general purpose needing only the name of the entity, I kind of didn't want to hurt its 'elegance' by having to do complicated switching over its type to extract the name after the fact, all so I could use the Ship to see if it was big|huge.

Reply:
Perhaps, but `mission_log_add_entry` does contain code that already gets the ship object.